code/modules/mob/inventory.dm

```diff
@@ -250,22 +254,24 @@ var/global/list/slot_equipment_priority = list( \
 	return TRUE
 
 //Attemps to remove an object on a mob.
-/mob/proc/remove_from_mob(var/obj/O, var/atom/target)
-	if(!O) // Nothing to remove, so we succeed.
-		return TRUE
-	if (src.client)
-		src.client.screen -= O
-	O.reset_plane_and_layer()
-	O.screen_loc = null
-	if(!src.u_equip(O))
+/mob/proc/remove_from_mob(var/obj/object, var/atom/target)
+	if(!object) // Nothing to remove, so we succeed.
 		return TRUE
-	if(istype(O, /obj/item))
-		var/obj/item/I = O
+
+	src.u_equip(object)
+	src.client?.screen -= object
+	object.reset_plane_and_layer()
+	object.screen_loc = null
+
+	if(istype(object, /obj/item))
+		var/obj/item/item_to_remove = object
 		if(target)
-			I.forceMove(target)
+			item_to_remove.forceMove(target)
```

Comment:
This should probably be dropInto, meaning this entire if should be combined into `item_to_remove.dropInto(target || loc)`.